src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8632,6 +8632,20 @@ public override BoundNode VisitEventAssignmentOperator(BoundEventAssignmentOpera
             }
             VisitRvalue(node.Argument);
             // https://github.com/dotnet/roslyn/issues/31018: Check for delegate mismatch.
+            if (node.Argument.ConstantValue?.IsNull != true)
```

Comment:
>if (node.Argument.ConstantValue?.IsNull != true) [](start = 12, length = 48)

Is it important that the argument is a constant `null`? (That seems like an unlikely scenario.) Would it make sense to simply use the nullable state of the argument?

In short, should `E += x; E.Invoke()` result in a warning?
```C#
event Action? E;
void M(Action? x)
{
    E += x;
    E.Invoke(); // warning?
}
void M2(Action y)
{
    E += y;
    E.Invoke();
}
```

Reply:
It's not an important scenario, it just represents how event assignment works for field-like events. Adding or removing a null is basically a no-op.

> In short, should E += x; E.Invoke(); result in a warning?
Yes.